libs/linglong/src/linglong/runtime/container.cpp

```diff
@@ -53,7 +53,27 @@ void mergeProcessConfig(ocppi::runtime::config::types::Process &dst,
     }
 
     if (src.env) {
-        dst.env = src.env;
+        if (!dst.env) {
+            dst.env = src.env;
+        } else {
+            auto &srcEnv = src.env.value();
+            auto &dstEnv = dst.env.value();
+            for (const auto &env : srcEnv) {
+                auto key = env.substr(0, env.find_first_of('='));
+                auto it =
+                  std::find_if(dstEnv.begin(), dstEnv.end(), [&key](const std::string &env) {
+                      return env.rfind(key + "=", 0) == 0;
+                  });
```

Comment:
auto key = env.find_first_not_of('=');
            if (key == std::string::npos) {
                continue;
            }

auto it = std::find_if(dstEnv.begin(), dstEnv.end(), [key](const std::string &env) {
                return env.rfind(std::string_view(env.data(), key), 0) == 0;
            });